apps/src/templates/MultipleSectionsAssigner.jsx

```diff
@@ -0,0 +1,346 @@
+    );
+    if (isUnchecked) {
+      this.setState(state => {
+        let newList = state.currentSectionsAssigned.filter(
+          s => s.code !== currentSection.code
+        );
+        return {currentSectionsAssigned: newList};
+      });
+    } else {
+      this.setState(state => {
+        let newList = [...state.currentSectionsAssigned];
+        newList.push(currentSection);
+        return {currentSectionsAssigned: newList};
+      });
+    }
+  };
+
+  reassignSections = () => {
+    const {
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId,
+      assignToSection,
+      isOnCoursePage
+    } = this.props;
+    // Assign any courses that need to be assigned
+    for (let i = 0; i < this.state.currentSectionsAssigned.length; i++) {
+      const needsToBeAssigned = !this.state.initialSectionsAssigned.some(
+        s => s.code === this.state.currentSectionsAssigned[i].code
+      );
+      if (needsToBeAssigned) {
+        if (isOnCoursePage) {
+          const sectionId = this.state.currentSectionsAssigned[i].id;
+          assignToSection(
+            sectionId,
+            courseId,
+            courseOfferingId,
+            courseVersionId,
+            scriptId
+          );
+        } else {
+          this.unhideAndAssignUnit(this.state.currentSectionsAssigned[i]);
+        }
+      }
+    }
+
+    // If any sections need to be removed from being assigned, remove them
+    for (let i = 0; i < this.state.initialSectionsAssigned.length; i++) {
+      const isSectionToBeRemoved = !this.state.currentSectionsAssigned.some(
+        s => s.code === this.state.initialSectionsAssigned[i].code
+      );
+
+      if (isSectionToBeRemoved) {
+        // if on COURSE landing page or a STANDALONE UNIT, unassign entirely
+        // note, I don't know of a better way to check if it is on a course landing page
+        isOnCoursePage || this.props.isStandAloneUnit
+          ? this.props.unassignSection(
+              this.state.initialSectionsAssigned[i].id,
+              ''
+            )
+          : this.assignCourseWithoutUnit(this.state.initialSectionsAssigned[i]);
+      }
+    }
+    // close dialogue
+    this.props.reassignConfirm();
+    this.props.onClose();
+  };
+
+  unhideAndAssignUnit = section => {
+    const {
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId,
+      assignToSection,
+      updateHiddenScript,
+      testingFunction
+    } = this.props;
+    const sectionId = section.id;
+    updateHiddenScript(sectionId, scriptId, false);
+    assignToSection(
+      sectionId,
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId
+    );
+    testingFunction(
+      sectionId,
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId
+    );
+  };
+
+  // this is the same as the above function but just has null as the scriptId
+  // not sure if this should be its own function or not
```

Comment:
Comments for these functions is great but can you clean them up a bit? Even something like "This function is very similar to the one above but uses a null scriptId" would be fine.